[connector/servicegraph] Unit test failure: TestConnectorConsume/test_fix_failed_label_not_work

### Component(s)

connector/servicegraph

### Describe the issue you're reporting

[Failing CI/CD action link
](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/9854192691/job/27233186145?pr=33458)

Failure occurred in unrelated PR, running on ubuntu.

Failure output:
```
Running target 'test-with-cover' in module 'connector/servicegraphconnector' as part of group 'connector'
make --no-print-directory -C connector/servicegraphconnector test-with-cover
mkdir -p /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/coverage/unit
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 600s -parallel 4 --tags="" -cover -covermode=atomic -args -test.gocoverdir="/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/coverage/unit"
✖  . (79ms) (coverage: 87.3% of statements)
✓  internal/metadata (1.02s) (coverage: 83.3% of statements)
✓  internal/store (1.121s) (coverage: 92.7% of statements)

DONE 31 tests, 2 failures in 7.887s

✖  . (38ms) (coverage: 87.3% of statements)

=== Failed
make[2]: *** [../../Makefile.Common:131: test-with-cover] Error 1
=== FAIL: . TestConnectorConsume/test_fix_failed_label_not_work (0.01s)
make[1]: *** [Makefile:179: connector/servicegraphconnector] Error 2
    logger.go:146: 2024-07-09T18:45:05.793Z	INFO	Started servicegraphconnector
    logger.go:146: 2024-07-09T18:45:05.793Z	INFO	Shutting down servicegraphconnector
    logger.go:146: 2024-07-09T18:45:05.796Z	DEBUG	edge completed	{"client_service": "foo", "server_service": "bar", "connection_type": "", "trace_id": "b3a53e9c2ac533afc48b511c2f368dc1"}
    logger.go:146: 2024-07-09T18:45:05.796Z	DEBUG	edge completed	{"client_service": "foo", "server_service": "bar", "connection_type": "", "trace_id": "c6aafb2dcf4d71c81ae52e50e8f805ee"}
make: *** [Makefile:131: gotest-with-cover] Error 2
    logger.go:146: 2024-07-09T18:45:05.796Z	DEBUG	edge completed	{"client_service": "foo", "server_service": "bar", "connection_type": "", "trace_id": "7af657de319d6a57967bc241882ea94f"}
    connector_test.go:129: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/connector/servicegraphconnector/connector_test.go:129
        	Error:      	Received unexpected error:
        	            	the following errors occurred:
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_client_seconds": missing expected datapoint: map[client:foo connection_type: failed:false server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_client_seconds": unexpected datapoint: map[client:foo connection_type: failed:true server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_client_seconds": missing expected datapoint: map[client:foo connection_type: failed:true server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_client_seconds": unexpected datapoint: map[client:foo connection_type: failed:false server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": missing expected datapoint: map[client:foo connection_type: failed:false server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": unexpected datapoint: map[client:foo connection_type: failed:true server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": missing expected datapoint: map[client:foo connection_type: failed:true server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": unexpected datapoint: map[client:foo connection_type: failed:false server:bar]
        	Test:       	TestConnectorConsume/test_fix_failed_label_not_work

=== FAIL: . TestConnectorConsume (0.01s)

=== FAIL: . TestConnectorConsume/test_fix_failed_label_not_work (re-run 1) (0.01s)
    logger.go:146: 2024-07-09T18:45:07.902Z	INFO	Started servicegraphconnector
    logger.go:146: 2024-07-09T18:45:07.902Z	INFO	Shutting down servicegraphconnector
    logger.go:146: 2024-07-09T18:45:07.904Z	DEBUG	edge completed	{"client_service": "foo", "server_service": "bar", "connection_type": "", "trace_id": "b3a53e9c2ac533afc48b511c2f368dc1"}
    logger.go:146: 2024-07-09T18:45:07.905Z	DEBUG	edge completed	{"client_service": "foo", "server_service": "bar", "connection_type": "", "trace_id": "c6aafb2dcf4d71c81ae52e50e8f805ee"}
    logger.go:146: 2024-07-09T18:45:07.905Z	DEBUG	edge completed	{"client_service": "foo", "server_service": "bar", "connection_type": "", "trace_id": "7af657de[319](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/9854192691/job/27233186145?pr=33458#step:9:320)d6a57967bc241882ea94f"}
    connector_test.go:129: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/connector/servicegraphconnector/connector_test.go:129
        	Error:      	Received unexpected error:
        	            	the following errors occurred:
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": missing expected datapoint: map[client:foo connection_type: failed:false server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": unexpected datapoint: map[client:foo connection_type: failed:true server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": missing expected datapoint: map[client:foo connection_type: failed:true server:bar]
        	            	 -  resource "map[]": scope "traces_service_graph": metric "traces_service_graph_request_server_seconds": unexpected datapoint: map[client:foo connection_type: failed:false server:bar]
        	Test:       	TestConnectorConsume/test_fix_failed_label_not_work

=== FAIL: . TestConnectorConsume (re-run 1) (0.01s)

DONE 2 runs, 33 tests, 4 failures in 8.913s
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
Error: Process completed with exit code 2.
```

--- a/connector/servicegraphconnector/connector_test.go
+++ b/connector/servicegraphconnector/connector_test.go
@@ -125,6 +125,7 @@ func TestConnectorConsume(t *testing.T) {
 			pmetrictest.IgnoreMetricDataPointsOrder(),
 			pmetrictest.IgnoreStartTimestamp(),
 			pmetrictest.IgnoreTimestamp(),
+			pmetrictest.IgnoreDatapointAttributesOrder(),
 		)
 		require.NoError(t, err)
 	})

--- a/pkg/pdatatest/pmetrictest/options.go
+++ b/pkg/pdatatest/pmetrictest/options.go
@@ -183,12 +183,12 @@ func IgnoreMetricAttributeValue(attributeName string, metricNames ...string) Com
 // IgnoreMetricAttributeValue is a CompareMetricsOption that clears value of the metric attribute.
 func IgnoreDatapointAttributesOrder() CompareMetricsOption {
 	return compareMetricsOptionFunc(func(expected, actual pmetric.Metrics) {
-		_ = orderDatapointAttributes(expected)
-		_ = orderDatapointAttributes(actual)
+		orderDatapointAttributes(expected)
+		orderDatapointAttributes(actual)
 	})
 }
 
-func orderDatapointAttributes(metrics pmetric.Metrics) error {
+func orderDatapointAttributes(metrics pmetric.Metrics) {
 	rms := metrics.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		ilms := rms.At(i).ScopeMetrics()
@@ -200,49 +200,33 @@ func orderDatapointAttributes(metrics pmetric.Metrics) error {
 				case pmetric.MetricTypeGauge:
 					for k := 0; k < msl.At(g).Gauge().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Gauge().DataPoints().At(k).Attributes().AsRaw())
-						err := msl.At(g).Gauge().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
-						if err != nil {
-							return err
-						}
+						_ = msl.At(g).Gauge().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
 					}
 				case pmetric.MetricTypeSum:
 					for k := 0; k < msl.At(g).Sum().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Sum().DataPoints().At(k).Attributes().AsRaw())
-						err := msl.At(g).Sum().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
-						if err != nil {
-							return err
-						}
+						_ = msl.At(g).Sum().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
 					}
 				case pmetric.MetricTypeHistogram:
 					for k := 0; k < msl.At(g).Histogram().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Histogram().DataPoints().At(k).Attributes().AsRaw())
-						err := msl.At(g).Histogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
-						if err != nil {
-							return err
-						}
+						_ = msl.At(g).Histogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
 					}
 				case pmetric.MetricTypeExponentialHistogram:
 					for k := 0; k < msl.At(g).ExponentialHistogram().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().AsRaw())
-						err := msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
-						if err != nil {
-							return err
-						}
+						_ = msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
 					}
 				case pmetric.MetricTypeSummary:
 					for k := 0; k < msl.At(g).Summary().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Summary().DataPoints().At(k).Attributes().AsRaw())
-						err := msl.At(g).Summary().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
-						if err != nil {
-							return err
-						}
+						_ = msl.At(g).Summary().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
 					}
 				case pmetric.MetricTypeEmpty:
 				}
 			}
 		}
 	}
-	return nil
 }
 
 func orderMapByKey(input map[string]any) map[string]any {

--- a/pkg/pdatatest/pmetrictest/options.go
+++ b/pkg/pdatatest/pmetrictest/options.go
@@ -183,12 +183,12 @@ func IgnoreMetricAttributeValue(attributeName string, metricNames ...string) Com
 // IgnoreMetricAttributeValue is a CompareMetricsOption that clears value of the metric attribute.
 func IgnoreDatapointAttributesOrder() CompareMetricsOption {
 	return compareMetricsOptionFunc(func(expected, actual pmetric.Metrics) {
-		orderDatapointAttributes(expected)
-		orderDatapointAttributes(actual)
+		_ = orderDatapointAttributes(expected)
+		_ = orderDatapointAttributes(actual)
 	})
 }
 
-func orderDatapointAttributes(metrics pmetric.Metrics) {
+func orderDatapointAttributes(metrics pmetric.Metrics) error {
 	rms := metrics.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		ilms := rms.At(i).ScopeMetrics()
@@ -200,33 +200,49 @@ func orderDatapointAttributes(metrics pmetric.Metrics) {
 				case pmetric.MetricTypeGauge:
 					for k := 0; k < msl.At(g).Gauge().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Gauge().DataPoints().At(k).Attributes().AsRaw())
-						msl.At(g).Gauge().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						err := msl.At(g).Gauge().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						if err != nil {
+							return err
+						}
 					}
 				case pmetric.MetricTypeSum:
 					for k := 0; k < msl.At(g).Sum().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Sum().DataPoints().At(k).Attributes().AsRaw())
-						msl.At(g).Sum().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						err := msl.At(g).Sum().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						if err != nil {
+							return err
+						}
 					}
 				case pmetric.MetricTypeHistogram:
 					for k := 0; k < msl.At(g).Histogram().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Histogram().DataPoints().At(k).Attributes().AsRaw())
-						msl.At(g).Histogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						err := msl.At(g).Histogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						if err != nil {
+							return err
+						}
 					}
 				case pmetric.MetricTypeExponentialHistogram:
 					for k := 0; k < msl.At(g).ExponentialHistogram().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().AsRaw())
-						msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						err := msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						if err != nil {
+							return err
+						}
 					}
 				case pmetric.MetricTypeSummary:
 					for k := 0; k < msl.At(g).Summary().DataPoints().Len(); k++ {
 						rawOrdered := orderMapByKey(msl.At(g).Summary().DataPoints().At(k).Attributes().AsRaw())
-						msl.At(g).Summary().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						err := msl.At(g).Summary().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+						if err != nil {
+							return err
+						}
 					}
 				case pmetric.MetricTypeEmpty:
 				}
 			}
 		}
 	}
+	return nil
 }
 
 func orderMapByKey(input map[string]any) map[string]any {

--- a/pkg/pdatatest/pmetrictest/options.go
+++ b/pkg/pdatatest/pmetrictest/options.go
@@ -7,6 +7,7 @@ import (
 	"bytes"
 	"fmt"
 	"regexp"
+	"sort"
 	"time"
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
@@ -177,6 +178,74 @@ func IgnoreMetricAttributeValue(attributeName string, metricNames ...string) Com
 		maskMetricAttributeValue(expected, attributeName, metricNames)
 		maskMetricAttributeValue(actual, attributeName, metricNames)
 	})
+}
+
+// IgnoreMetricAttributeValue is a CompareMetricsOption that clears value of the metric attribute.
+func IgnoreDatapointAttributesOrder() CompareMetricsOption {
+	return compareMetricsOptionFunc(func(expected, actual pmetric.Metrics) {
+		orderDatapointAttributes(expected)
+		orderDatapointAttributes(actual)
+	})
+}
+
+func orderDatapointAttributes(metrics pmetric.Metrics) {
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		ilms := rms.At(i).ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			msl := ilms.At(j).Metrics()
+			for g := 0; g < msl.Len(); g++ {
+				msl.At(g)
+				switch msl.At(g).Type() {
+				case pmetric.MetricTypeGauge:
+					for k := 0; k < msl.At(g).Gauge().DataPoints().Len(); k++ {
+						rawOrdered := orderMapByKey(msl.At(g).Gauge().DataPoints().At(k).Attributes().AsRaw())
+						msl.At(g).Gauge().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+					}
+				case pmetric.MetricTypeSum:
+					for k := 0; k < msl.At(g).Sum().DataPoints().Len(); k++ {
+						rawOrdered := orderMapByKey(msl.At(g).Sum().DataPoints().At(k).Attributes().AsRaw())
+						msl.At(g).Sum().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+					}
+				case pmetric.MetricTypeHistogram:
+					for k := 0; k < msl.At(g).Histogram().DataPoints().Len(); k++ {
+						rawOrdered := orderMapByKey(msl.At(g).Histogram().DataPoints().At(k).Attributes().AsRaw())
+						msl.At(g).Histogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+					}
+				case pmetric.MetricTypeExponentialHistogram:
+					for k := 0; k < msl.At(g).ExponentialHistogram().DataPoints().Len(); k++ {
+						rawOrdered := orderMapByKey(msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().AsRaw())
+						msl.At(g).ExponentialHistogram().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+					}
+				case pmetric.MetricTypeSummary:
+					for k := 0; k < msl.At(g).Summary().DataPoints().Len(); k++ {
+						rawOrdered := orderMapByKey(msl.At(g).Summary().DataPoints().At(k).Attributes().AsRaw())
+						msl.At(g).Summary().DataPoints().At(k).Attributes().FromRaw(rawOrdered)
+					}
+				case pmetric.MetricTypeEmpty:
+				}
+			}
+		}
+	}
+}
+
+func orderMapByKey(input map[string]any) map[string]any {
+	// Create a slice to hold the keys
+	keys := make([]string, 0, len(input))
+	for k := range input {
+		keys = append(keys, k)
+	}
+
+	// Sort the keys
+	sort.Strings(keys)
+
+	// Create a new map to hold the sorted key-value pairs
+	orderedMap := make(map[string]any, len(input))
+	for _, k := range keys {
+		orderedMap[k] = input[k]
+	}
+
+	return orderedMap
 }
 
 func maskMetricAttributeValue(metrics pmetric.Metrics, attributeName string, metricNames []string) {

